Code/GraphMol/Chirality.cpp

```diff
@@ -881,17 +882,18 @@ void buildCIPInvariants(const ROMol &mol, DOUBLE_VECT &res) {
   }
 }
 
-void iterateCIPRanks(const ROMol &mol, DOUBLE_VECT &invars, UINT_VECT &ranks,
-                     bool seedWithInvars) {
+void iterateCIPRanks(const ROMol &mol, const DOUBLE_VECT &invars,
+                     UINT_VECT &ranks, bool seedWithInvars) {
   PRECONDITION(invars.size() == mol.getNumAtoms(), "bad invars size");
   PRECONDITION(ranks.size() >= mol.getNumAtoms(), "bad ranks size");
 
   unsigned int numAtoms = mol.getNumAtoms();
   CIP_ENTRY_VECT cipEntries(numAtoms);
-  INT_LIST allIndices;
-  for (unsigned int i = 0; i < numAtoms; ++i) {
-    allIndices.push_back(i);
+  for (auto& vec : cipEntries) {
+    vec.reserve(16);
   }
+  INT_LIST allIndices(numAtoms);
```

Comment:
Good point, I pushed a second commit to this PR to just create normal for loops for `index` instead of materializing the vector.  Runtime of `iterateCIPRanks` improves by about 10% with that second change.